Tests : activating end-to-end tests on Windows OS

guilpier-code opened the pull request.

guilpier-code committed:
Tests : activating end-to-end tests on Windows OS

guilpier-code committed:
Making tests run output more readable

guilpier-code committed:
Allow pytest to display logs from print(...)

guilpier-code committed:
Test too see if we display details on errors

guilpier-code committed:
Refactoring code for comparison of simulation to reference, at least to remove this horrible recursive compare function.

flomnes commented on src/tests/examples/antares_test_utils.py:
```diff
@@ -90,46 +58,118 @@ def enable_study_output(study_path, enable):
+    return False
+
+def skip_file(file):
+    if file in ['id-daily.txt', 'id-hourly.txt']:
+        return True
+    return False
```
`return file in ['id-daily.txt', 'id-hourly.txt']`

guilpier-code replied:
As the function is named **skip_file**, it may return **true** or **false**. returning anything else could be confusing for the caller

flomnes replied:
```python
>>> 1 in [1,2,3]
True
>>> 4 in [1,2,3]
False
```

guilpier-code replied:
Sorry. Done.

guilpier-code committed:
Comparison report improvements (see details after)

- Removing report's embarrassing error logs from a study execution
- Reducing report's number of lines per file to be printed down to 10
- Changing the report columns name (self ==> reference, other ==> results)
- Adding a column to the report : the differences between the 2 first columns

guilpier-code committed:
Run tests on CI : fixing 'diff' column values (relevant values are the opposite)

guilpier-code committed:
Run tests on CI : make possible to set tolerances for each test

flomnes commented on src/tests/examples/antares_test_utils.py:
```diff
@@ -90,46 +58,132 @@ def enable_study_output(study_path, enable):
+    elif sys.platform == "win32":
+        return Win_tolerances()
+    else:
+        print("Unknown OS")
+        raise AssertionError()
```
raise AssertionError("Unknown OS")

guilpier-code replied:
Indeed, that would be better. But option **--tb=no** when calling **output_compare_test.py** prevents any trace back report for assertions to be printed.
We could use  **--tb=line** instead (one line of trace back per failure).
But in that case we have a report like this (2 tests are failing) : 
```
================================== FAILURES ===================================
D:\Users\pierregui\Desktop\ant-open-rel\src\tests\examples\antares_test_utils.py:178: AssertionError: Comparison failed
D:\Users\pierregui\Desktop\ant-open-rel\src\tests\examples\antares_test_utils.py:178: AssertionError: Comparison failed
=========================== short test summary info ===========================
FAILED output_compare_test.py::test_065_pumped_storage_plant_explicit_model_01
FAILED output_compare_test.py::test_067_pumped_storage_plant_explicit_model_03
================= 2 failed, 1 passed, 82 deselected in 34.11s =================
```
We have one trace back per failure, but failed assertions are not associated to a test. So we could insert the test name in the assertion message.

flomnes commented on src/tests/examples/output_compare_test.py:
```diff
@@ -7,139 +7,158 @@
 ## TESTS ##
 @pytest.mark.short
 def test_001_one_node_passive(use_ortools, ortools_solver, solver_path):
+    print()
+    print("Running the tests")
```
Remove this print

guilpier-code replied:
Done

flomnes commented on src/tests/examples/output_compare_test.py:
```diff
@@ -513,95 +597,107 @@ def test_071_hydro_reservoir_model_enhanced_04(use_ortools, ortools_solver, solv
     run_study(solver_path, study_path, use_ortools, ortools_solver)
     enable_study_output(study_path, False)
-    check_output_values(study_path)
+    tolerances = get_tolerances()
+    check_output_values(study_path, tolerances)
```
`check_output_values(study_path, get_tolerances())`

guilpier-code replied:
ok, done

guilpier-code committed:
Run tests on CI : set tolerances for 2 tests

guilpier-code committed:
Run tests on CI : allow sending std error to devnull for unfeasible problem tests

guilpier-code committed:
Run tests on CI : corrections due to code review

guilpier-code committed:
Run tests on CI : add an enlarged tolerance to test "playlist-psp-misc-ndg"
